rpc/src/helpers.rs

```diff
@@ -164,20 +164,20 @@ impl BlockHeaderShellInfo {
                 .map(|x| hex::encode(&x))
                 .collect(),
             context: block.header.context().to_base58_check(),
-        }
+        })
     }
 }
 
 impl BlockHeaderInfo {
-    pub fn new(
+    pub fn try_new(
         block: &BlockHeaderWithHash,
         block_json_data: &BlockJsonData,
         block_additional_data: &BlockAdditionalData,
         chain_id: &ChainId,
-    ) -> Self {
+    ) -> Result<Self, TimestampOutOfRangeError> {
         let header: &BlockHeader = &block.header;
         let predecessor = header.predecessor().to_base58_check();
-        let timestamp = ts_to_rfc3339(header.timestamp());
+        let timestamp = ts_to_rfc3339(header.timestamp())?;
```

Comment:
I'm not sure if that is possible, but please check if it would be easier to perform the conversion on use, not on construction.